src/Headphones.h

```diff
@@ -206,23 +331,114 @@ class Headphones
-	HeadphonesEvent _handleNcAsmParam(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleProtocolInfo(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleCapabilityInfo(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleDeviceInfo(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleT1SupportFunction(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleT2SupportFunction(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleNcAsmParam(const HeadphonesMessage &msg, CommandType ct);
 	HeadphonesEvent _handleBatteryLevelRet(const HeadphonesMessage &msg);
 	HeadphonesEvent _handlePlaybackStatus(const HeadphonesMessage &msg);
 	HeadphonesEvent _handlePlaybackSndPressureRet(const HeadphonesMessage &msg);
+	HeadphonesEvent _handlePowerParam(const HeadphonesMessage &msg, CommandType ct);
 	HeadphonesEvent _handleVoiceGuidanceParam(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleMultipointDevice(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleConnectedDevices(const HeadphonesMessage &msg);
 	HeadphonesEvent _handlePlaybackStatusControl(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleMultipointEtcEnable(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleListeningMode(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleAutomaticPowerOffButtonMode(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleSpeakToChat(const HeadphonesMessage &msg);
+	HeadphonesEvent _handleEqualizerAvailable(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleEqualizer(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleMiscDataRet(const HeadphonesMessage &msg);
 	HeadphonesEvent _handleMessage(HeadphonesMessage const &msg);
 
 	bool hasInit = false;
 };
+
+template <typename, typename = void>
+struct PayloadIsForMultipleCommandTypes : std::false_type {};
+
+template <typename T>
+struct PayloadIsForMultipleCommandTypes<T, std::void_t<decltype(T::COMMAND_IDS)>> : std::true_type {};
+
+template <typename T>
+inline constexpr bool PayloadIsForMultipleCommandTypes_v = PayloadIsForMultipleCommandTypes<T>::value;
+
+template <typename TPayload, typename... TArgs>
+void Headphones::sendGet(TArgs&&... args)
+{
+	if constexpr (PayloadIsForMultipleCommandTypes_v<TPayload>) {
+		static_assert(static_cast<uint8_t>(TPayload::COMMAND_IDS[CT_Get]) != 0xFF, "This command does not support Get");
+	}
+	return sendMdrCommandWithType<TPayload>(CT_Get, std::forward<TArgs>(args)...);
+}
+
+template <typename TPayload, typename... TArgs>
+void Headphones::sendSet(TArgs&&... args)
+{
+	if constexpr (PayloadIsForMultipleCommandTypes_v<TPayload>) {
+		static_assert(static_cast<uint8_t>(TPayload::COMMAND_IDS[CT_Set]) != 0xFF, "This command does not support Set");
+	}
+	return sendMdrCommandWithType<TPayload>(CT_Set, std::forward<TArgs>(args)...);
+}
+
+template <typename TPayload, typename... TArgs>
+void Headphones::sendMdrCommandWithType(CommandType ct, TArgs&&... args)
+{
+	if constexpr (TPayload::VARIABLE_SIZE) {
```

Comment:
I see there's `StringData`, `ConnectRetSupportFunction` payloads that's of variable sizes.  (curious - do we ever _send_ such payloads to the headphones?)  
Considering we'd allocate on the heap anyways, the `unique_ptr` indirection is uncessary - I suggest implementing dedicated serialization-to-bytes fucntions for these as there's no real benefit to _always_ alias the underlying data.